linkedList.js

```diff
@@ -0,0 +1,372 @@
+    return length;
+  }
+
+  insert(value) {
+    var node = new Node(value);
+    var currentNode = this.head;
+
+    if(!currentNode) {
+      this.head = node;
+
+      return node;
+    }
+
+    currentNode = this.head;
+    node.nextNode = currentNode;
+    this.head = node;
+
+    this._length++;
+    return node;
+  }
+
+  search(value) {
+    var currentNode = this.head;
+    if(!currentNode) {
+      return false;
+    }
+
+    while(currentNode.nextNode) {
+      if(currentNode.getData() === value) {
+        return true;
+      }
+      currentNode = currentNode.nextNode;
+    }
+
+    return false;
+  }
+
+  findMax() {
+    var currentNode = this.head;
+    var max = this.head.getData();
+    if(!currentNode) {
+      return null;
+    }
+
+    for(var i = 0; i < this.length(); i++) {
+      if(currentNode.getData() > max) {
+        max = currentNode.getData();
+      }
+      currentNode = currentNode.nextNode;
+    }
+
+    return max;
+  }
+
+  findMin() {
+    var currentNode = this.head;
+    var min = this.head.getData();
+
+    if(!currentNode) {
+      return null;
+    }
+
+    for(var i = 0; i < this.length(); i++) {
+      if(currentNode.getData() < min) {
+        min = currentNode.getData();
+      }
+      currentNode = currentNode.nextNode;
+    }
+
+    return min;
+  }
+
+  findNthFromBeginning(index) {
+    var currentNode = this.head;
+    if(!currentNode) {
+      return null;
+    }
+    for(var i = 0; i < index; i++) {
+      currentNode = currentNode.nextNode;
+    }
+
+    return (currentNode ? currentNode.getData() : null);
+  }
+
+  visit() {
+    var printString = "";
+    var currentNode = this.head;
+    if(!currentNode) {
+      return null;
+    }
+
+    for(var i = 0; i < this.length(); i++) {
+      printString += currentNode.getData() + " ";
+      currentNode = currentNode.nextNode;
+    }
+
+    console.log(printString);
+  }
+
+  insertAscending(value) {
+    var currentNode = this.head;
+    var node = new Node(value);
+    var inserted = false;
+
+
+
+    if(!currentNode) {
+      return null;
+    }
+
+    while(currentNode.nextNode) {
+      var nextNode = currentNode.nextNode;
+
+      if(currentNode.getData() < value && nextNode.getData() > value) {
+        var tempNode = currentNode.nextNode;
+        node.nextNode = tempNode;
+        currentNode.nextNode = node;
+        inserted = true;
+
+      }
+      currentNode = currentNode.nextNode;
+    }
+
+    if(!inserted) {
+      currentNode.nextNode = node;
+    }
+  }
+
+  deleteNode(value) {
+    var currentNode = this.head;
+    var nextNode;
+
+    if(!currentNode) {
+      return null;
+    }
+
+    if(this.head.getData() === value ) {
+      nextNode = this.head.nextNode;
+      this.head = nextNode;
+    } else {
+      while(currentNode.nextNode) {
+        nextNode = currentNode.nextNode;
+        if(nextNode.getData() === value ) {
+          var tempNode = nextNode.nextNode;
+          currentNode.nextNode = nextNode.nextNode;
```

Comment:
You can safely delete line 174. Also, you can return after updating nextNode and deleting the node.